lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -89,15 +89,9 @@ def compile(node)
 
     benchmark(:notice, str) do
       Puppet::Util::Profiler.profile(str, [:compiler, :compile, node.environment, node.name]) do
-        begin
-          config = Puppet::Parser::Compiler.compile(node)
-        rescue Puppet::Error => detail
-          Puppet.err(detail.to_s) if networked?
-          raise
-        end
```

Comment:
The only statement within begin/rescue here was a call to `Puppet::Parser::Compile.compile(node)` (the class method). That method will _always_ rescue, log, and re-raise all exceptions. So what this change does is avoiding a duplicate from being logged when networked.